Mathlib/MeasureTheory/Function/SimpleFunc.lean

```diff
@@ -553,6 +553,49 @@ instance instModule [Semiring K] [AddCommMonoid β] [Module K β] : Module K (α
 theorem smul_eq_map [SMul K β] (k : K) (f : α →ₛ β) : k • f = f.map (k • ·) :=
   rfl
 
+instance [NonUnitalNonAssocSemiring β] : NonUnitalNonAssocSemiring (α →ₛ β) where
+  left_distrib _ _ _ := ext fun _ ↦ left_distrib ..
+  right_distrib _ _ _ := ext fun _ ↦ right_distrib ..
+  zero_mul _ := ext fun _ ↦ zero_mul ..
+  mul_zero _ := ext fun _ ↦ mul_zero ..
+
+instance [Semiring β] : Semiring (α →ₛ β) where
```

Comment:
There are other ring-like structures to include here. `NonUnitalSemiring`, all the non-semi variants. Commutative versions.